### PR TITLE
fix: update workstation hour rate when workstation change in job card

### PR DESCRIPTION
**Pull request by Vishnu7025**
fix: update workstation hour rate when workstation change in job card

### CODE REVIEW

File: erpnext/manufacturing/doctype/job_card/job_card.py [hunk 1]
```diff
@@ -547,6 +547,7 @@ def validate_produced_quantity(self, for_quantity, wo):
 			)
 
 	def update_work_order_data(self, for_quantity, time_in_mins, wo):
+		ws_hr_rate = frappe.get_value("Workstation", self.workstation, "hour_rate")
```
Comment (marination) [inline erpnext/manufacturing/doctype/job_card/job_card.py]:
```suggestion
		workstation_hour_rate = frappe.get_value("Workstation", self.workstation, "hour_rate")
```
Explicit variable names are a must for readability

File: erpnext/manufacturing/doctype/job_card/job_card.py [hunk 2]
```diff
@@ -572,6 +573,7 @@ def update_work_order_data(self, for_quantity, time_in_mins, wo):
 				if data.get("workstation") != self.workstation:
 					# workstations can change in a job card
 					data.workstation = self.workstation
+					data.hour_rate = flt(ws_hr_rate)
```
Comment (marination) [inline erpnext/manufacturing/doctype/job_card/job_card.py]:
```suggestion
					data.hour_rate = flt(workstation_hour_rate)
```

**Review by marination — APPROVED** [review]

**Comment by Vishnu7025** [comment]
@rohitwaghchaure

**Comment by rohitwaghchaure** [comment]
@Mergifyio backport version-14-hotfix

**Comment by rohitwaghchaure** [comment]
@Vishnu7025 backport not working for this PR, could you open the PR on version-14-hotfix branch.

**Pull request merged**